Spine/Networking.swift

```diff
@@ -8,35 +8,43 @@
 
 import Foundation
 
-/// Callback used by the _HTTPClientProtocol
-typealias HTTPClientCallback = (statusCode: Int?, responseData: NSData?, networkError: NSError?) -> Void
+public typealias NetworkClientCallback = (statusCode: Int?, data: NSData?, error: NSError?) -> Void
 
 /**
-The HTTPClientProtocol declares methods and properties that a HTTP client must implement.
+A NetworkClient is the interface between Spine and the server. It does not impose any transport
+and can be used for HTTP, websockets, and any other data transport.
 */
-public protocol HTTPClientProtocol {
-	/// Sets a HTTP header for all upcoming network requests.
-	func setHeader(header: String, to: String)
+public protocol NetworkClient {
+	/**
+	Performs a network request to the given URL with the given method.
 	
-	/// Removes a HTTP header for all upcoming  network requests.
-	func removeHeader(header: String)
+	- parameter method:   The method to use, expressed as a HTTP verb.
+	- parameter URL:      The URL to which to make the request.
+	- parameter callback: The callback to execute when the request finishes.
+	*/
+	func request(method: String, URL: NSURL, callback: NetworkClientCallback)
+	
+	/**
+	Performs a network request to the given URL with the given method.
+	
+	- parameter method:   The method to use, expressed as a HTTP verb.
+	- parameter URL:      The URL to which to make the request.
+	- parameter payload:  The payload the send as part of the request.
+	- parameter callback: The callback to execute when the request finishes.
+	*/
+	func request(method: String, URL: NSURL, payload: NSData?, callback: NetworkClientCallback)
 }
 
-/**
-The _HTTPClientProtocol declares methods and properties that a HTTP client must implement.
-*/
-protocol _HTTPClientProtocol: HTTPClientProtocol {
-	/// Performs a network request to the given URL with the given HTTP method.
-	func request(method: String, URL: NSURL, callback: HTTPClientCallback)
-	
-	/// Performs a network request to the given URL with the given HTTP method and request body data.
-	func request(method: String, URL: NSURL, payload: NSData?, callback: HTTPClientCallback)
+extension NetworkClient {
+	public func request(method: String, URL: NSURL, callback: NetworkClientCallback) {
+		return request(method, URL: URL, payload: nil, callback: callback)
+	}
 }
 
 /**
-The built in HTTP client that uses an NSURLSession for networking.
+The HTTPClient implements the NetworkClient protocol to work over an HTTP connection.
 */
-public class URLSessionClient: _HTTPClientProtocol {
+public class HTTPClient: NetworkClient {
```

Comment:
Researching the code, I found out how to define headers by injecting this into `Spine()`, but somehow HTTPClient is not being accessible. Check this out:

<img width="791" alt="nohttpclient" src="https://cloud.githubusercontent.com/assets/70312/10713497/eb2c3a24-7a87-11e5-8c57-4eaccf6c47b8.png">